lib/THC/THCTensor.c

```diff
@@ -599,7 +599,7 @@ void THCudaTensor_freeCopyTo(THCState *state, THCudaTensor *self, THCudaTensor *
 static void THCudaTensor_rawInit(THCState *state, THCudaTensor *self)
 {
   self->refcount = 1;
-  self->storage = NULL;
+  self->storage = THCudaStorage_new(state);
```

Comment:
That sounds good to me. I'm just a bit worried to diverge from Torch's behaviour, but maybe it's not a problem

Re the return value of THCudaTensor_data, I'm not sure myself why it seems to behave differently. But without your patch calling `torch.CudaTensor():data` from Lua returns `nil`, whereas with your patch it returns `cdata<float *>: NULL`. Those two values behave differently in conditionals, so if you have `if x:data() then ...` in your code, the behaviour will now change for empty tensors.